### ISSUE TITLE
Strange behaviour between emulator app and actual device app

### ISSUE THREAD

**Issue by spg99** [Baseflow/flutter-geolocator#625]
## 🐛 Bug Report

Test app behaves differently on emulator versus actual device

### Expected behavior: Simple app to show current gps location and distance to hard coded coordinates should display current coordinates, and calculated distance. Works as expected on emulator, with generated location. When debugging on actual device, when the get location button is pressed, the getlocation call times out. Different behaviour occurs when I run a mock gps app on the actual device, the get location call returns with the mocked location! This is a good thing, because the ultimate goal is to get location information from a bluetooth gps device, that is using the mock location setting to replace the android device's native gps location.

(This is my first bug/issue report - please forgive me if guidelines not followed too closely)

### Reproduction steps: 
- run app on emulator and get expected location based on set emulator location;
- run app on actual device and get call timeout from getlocation call;
- run app on actual device with mock gps app running, and get the mocked location from getlocation call. 

### Configuration
- develop machine: Windows 10 with latest updates applied
- software: Visual Studio Code with latest updates applied, (see flutter doctor listing)

**Version:** 6.1.13 used
[New Text Document.txt](https://github.com/Baseflow/flutter-geolocator/files/5675482/New.Text.Document.txt)

**Platform:** 
- [ ] :iphone: iOS
- [x ] :robot: Android

**Comment by mvanbeusekom** [comment]
@spg99 this is possible if the real device doesn't have a solid GPS connection (e.g. you are debugging it indoors). When using the emulator and mock gps app a geolocation update is simulated and will always arrive. In your case you are forcing the use of the legacy `LocationManager` (by passing the `forceAndroidLocationManager: true` parameter) which when using the actual device and a `LocationAccuracy.high` (or higher) the device relies on the GPS signal to provide reliable location updates. When working indoors a GPS signal is often not available which results in a timeout.

If you can use the `FusedLocationProviderClient` instead of the `LocationManager` (by setting the `forceAndroidLocationManager` parameter to `false`) Android will try to determine the location based on a combination of GPS, WiFi and Cellular Network, which will often also result in a fix while being indoor.

So, in short this is expected behaviour when using the legacy `LocationManager` in situations when the GPS signal is weak (i.e. when indoors, dense woods or dense urban area's).

I will be closing this issue for now, but don't hesitate if you have any follow up questions.
